apps/api/src/app/notification-groups/usecases/delete-notification-group/delete-notification-group.usecase.ts

```diff
@@ -0,0 +1,32 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { NotificationGroupRepository, DalException } from '@novu/dal';
+import { DeleteNotificationGroupCommand } from './delete-notification-group.command';
+import { ApiException } from '../../../shared/exceptions/api.exception';
+@Injectable()
+export class DeleteNotificationGroup {
+  constructor(private notificationGroupRepository: NotificationGroupRepository) {}
+
+  async execute(command: DeleteNotificationGroupCommand) {
+    try {
+      const group = await this.notificationGroupRepository.findOne({
+        _organizationId: command.organizationId,
+      });
```

Comment:
I'm a bit confused of whule this query is needed, it will return random parentId's each time. I think it's better to remove it altogether from here and only delete using `command.id` not sure that parentId should be factored in